jetty-core/jetty-compression/jetty-compression-api/src/main/java/org/eclipse/jetty/compression/Compression.java

```diff
@@ -0,0 +1,355 @@
+     * </p>
+     *
+     * @return the list of common extension names (all lowercase) for this compression implementation.
+     * ordered by most common to least common.
+     */
+    public abstract List<String> getFileExtensionNames();
+
+    public abstract int getMinCompressSize();
+
+    public abstract void setMinCompressSize(int minCompressSize);
+
+    /**
+     * @return the name of the compression implementation.
+     */
+    public abstract String getName();
+
+    /**
+     * The {@link HttpField} for {@code X-Content-Encoding} suitable for this Compression implementation.
+     *
+     * @return the HttpField for {@code X-Content-Encoding}.
+     */
+    public abstract HttpField getXContentEncodingField();
+
+    /**
+     * Create a new {@link InputStream} to decode (decompress) with this compression implementation.
+     *
+     * @param in the input stream to write the decoded (decompressed) bytes to
+     * @return the {@link InputStream} implementation for this compression.
+     * @throws IOException if unable to create InputStream
+     */
+    public InputStream newDecoderInputStream(InputStream in) throws IOException
+    {
+        return newDecoderInputStream(in, getDefaultDecoderConfig());
+    }
+
+    /**
+     * Create a new {@link InputStream} to decode (decompress) with this compression implementation.
+     *
+     * @param in the input stream to write the decoded (decompressed) bytes to
+     * @param config the {@link DecoderConfig} for this input stream.
+     * @return the {@link InputStream} implementation for this compression.
+     * @throws IOException if unable to create InputStream
+     */
+    public abstract InputStream newDecoderInputStream(InputStream in, DecoderConfig config) throws IOException;
+
+    /**
+     * Create a new {@link DecoderSource} for this compression implementation
+     *
+     * @param source the source to write the decoded bytes to
+     * @return a new {@link DecoderSource}
+     */
+    public DecoderSource newDecoderSource(Content.Source source)
+    {
+        return newDecoderSource(source, getDefaultDecoderConfig());
+    }
+
+    /**
+     * Create a new {@link DecoderSource} for this compression implementation
+     *
+     * @param source the source to write the decoded bytes to
+     * @param config the {@link DecoderConfig} for this source.
+     * @return a new {@link DecoderSource}
+     */
+    public abstract DecoderSource newDecoderSource(Content.Source source, DecoderConfig config);
+
+    /**
+     * Create a new {@link OutputStream} to encode (compress) with this compression implementation.
+     *
+     * @param out the output stream to write the encoded (compressed) bytes to
+     * @return the {@link OutputStream} implementation for this compression.
+     * @throws IOException if unable to create OutputStream
+     */
+    public OutputStream newEncoderOutputStream(OutputStream out) throws IOException
```

Comment:
Ditto, from above ... This uses the compression implementation version of the OutputStream, not the wrapped version of a wrapped of an extension.  It's mostly used for tests and demonstration purposes.  It will come to be especially handy for Jetty Client users in the future (when this is integrated there) as well.